src/main/java/marquez/service/models/DatasetVersion.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package marquez.service.models;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import java.time.Instant;
+import javax.annotation.Nullable;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.ToString;
+import marquez.common.models.DatasetId;
+import marquez.common.models.DatasetName;
+import marquez.common.models.DatasetType;
+import marquez.common.models.Field;
+import marquez.common.models.NamespaceName;
+import marquez.common.models.RunId;
+import marquez.common.models.SourceName;
+import marquez.common.models.TagName;
+import marquez.common.models.Version;
+
+@EqualsAndHashCode
+@ToString
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.NAME,
+    include = JsonTypeInfo.As.EXISTING_PROPERTY,
+    property = "type")
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = DbTable.class, name = "DB_TABLE"),
+  @JsonSubTypes.Type(value = Stream.class, name = "STREAM")
+})
+public abstract class DatasetVersion {
```

Comment:
should we use the lombok @Delegate annotation? DatasetVersion is Dataset + a version UUID

Reply:
The object does differ enough where I think a new model make sense. For example, a version has a `createdByRun` that will return metadata about the run that created the version, versions are also immutable so an `updatedAt` field won't be part of the response model.

That said, using `@Delegate` would be a really nice approach if you'd be able to filter with delegate methods get generate.